src/libs/ReportUtils.ts

```diff
@@ -5415,7 +5415,10 @@ function shouldReportBeInOptionList({
             !isSelfDM(report) &&
             !isSystemChat(report) &&
             !isGroupChat(report) &&
-            !isInvoiceRoom(report))
+            !isInvoiceRoom(report) &&
+            // We omit sending back participants for chat rooms when searching for reports since they aren't needed to display the results and can get very large.
+            // So we allow showing rooms with no participants–in any other circumstances we should never have these reports with no participants in Onyx.
```

Comment:
Sorry to keep blocking here - but doesn't adding `!isChatRoom()` here make at least a few of the calls above redundant? I'm looking for the answers to these questions: 

1. Why do we have this condition and early return on main at the moment?
2. Which reports specifically are we allowing to display by adding`isChatRoom()`?

**Previous logic:**

```js
participantAccountIDs.length === 0 &&
    !isChatThread(report) &&
    !isPublicRoom(report) &&
    !isUserCreatedPolicyRoom(report) &&
    !isArchivedRoom(report) &&
    !isMoneyRequestReport(report) &&
    !isTaskReport(report) &&
    !isSelfDM(report) &&
    !isSystemChat(report) &&
    !isGroupChat(report) &&
    !isInvoiceRoom(report)
```

**Proposed logic:** 

all of those conditions + `!isChatRoom()` - but a few of them _are_ chat rooms, right? 🤔 

If we don't add this what bad thing will happen?

Reply:
Good point, I think we can actually remove the duplicate checks here, didn't think this through when I added it initially, good catch. 

We are allowing [the following reports](https://github.com/Expensify/App/blob/jasper-defaultRoomsEmptyParticipantsChatFinder/src/libs/ReportUtils.ts#L968-L973) with empty participants to display since their participants may be empty since they aren't returned from `SearchForReports`

I'll remove the duplicate checks.

Reply:
We have the following report types in NewDot currently, the ones being let through the condition are marked with an ❌
- **chat**
- expense ❌
- iou ❌
- task ❌
- **invoice**
- archived reports ❌

Of the chat reports, here are all the ones in NewDot, and I marked the ones being let through the condition
- **DMs**
- user-created policy rooms ❌
- default rooms (#announce, #admins, #domain.com) ❌
- trip rooms ❌
- invoice rooms ❌
- **policy expense chats**
- all chat threads (i.e. child reports with parent report type chat) ❌
- self-DMs ❌
- system chats ❌
- group chats ❌

So it seems like we are really only blocking the following report types with empty participants:
- DM chats
- policy expense chats
- invoice reports

So it seems like we could greatly simplify this condition, but I guess this brings us back to the question of whether we want reports to be opted-in or opted-out of this check if they have empty participants. To me it feels fine to let empty participant reports through in most cases.

Reply:
Thanks for the investigation - and yeah, in this case I think we should just be checking the report types that MUST have participants. 

But, tbh I'm still stuck on:

>>Why do we have this condition and early return on main at the moment?

>no idea

What would it take to get to the bottom of that? I am having trouble seeing what bad thing will happen if we just remove this condition entirely. Why would any of these reports have "no participants"? And if they don't, then should we quietly remove them as options? It seems like it would indicate some issue elsewhere that needs to be addressed.

Reply:
> then should we quietly remove them as options? It seems like it would indicate some issue elsewhere that needs to be addressed.

hmm totally agree with that, this definitely seems like a band-aid fix for some underlying back-end/data issues that should never happen.

I'm not sure that I should remove this check in my PR since it seems out of scope. Maybe we can add some type of ensure bugbot log for whenever this happens? Although this log may happen A LOT considering how this code is called 🤔

Reply:
I think it is a good idea to add a log for that particular condition (participants length + chatType) and create an issue to keep an eye on it, but NAB I think